lib/qxcli/commands/Command.js

```diff
@@ -42,6 +42,60 @@ qx.Class.define("qxcli.commands.Command", {
   members: {
     argv: null,
 
+    /**
+     * Returns the  path to the current library, depending on the current
+     * working directory. A library is assumed if a Manifest.json file
+     * exists. 
+     * @throws {Error} Throws an error if no library can be found.
+     * @return {String} The absolute path to the library
+     */ 
+    getLibraryPath: function(){ 
+      let libPath = process.cwd(); 
+      if ( ! fs.existsSync( path.join( libPath, "Manifest.json" ) ) ){
+        throw new Error( "Cannot find library - are you in the right directory?");
+      }
+      return libPath;
+    },
+
+    /**
+     * Returns the content of a json file in the library as an object. 
+     * @param {String} Relative path of the file in the library
+     * @throws {Error} Throws if library cannot be found or file cannot be parsed.
+     * @return {Object}
+     */
+    parseJsonFile : function(relativePath){
```

Comment:
This would be better done as an async function so the caller can decide whether to work synchronously

Reply:
Can you explain? Why would you turn a synchronous function async when there is no async action involved? The possibility of overriding it in a subclass?

Reply:
`parseJsonFile` calls `fs.readFileSync`, I think it's good practice to avoid sync functions wherever possible.  There is a `qx.utils.Promisify` which makes promisifying entire packages easier, you just wrap `require`, eg `var myModule = promisifyAll(require("myModule"));` and it adds an async version of every function in that module by appending "Async" to the function name.

Because the "fs" module needs special treatment, it's already available as a static, eg:

```
const fs = qx.utils.Promisify.fs;

async function simpleJsonReader(path) {
  return fs.readFileAsync(path, "utf8")
    .then(text => JSON.parse(text));
};

var obj = await simpleJsonReader("abc.json");
```

Reply:
You are generally right but so far, my impression hasn't been that using filesystem functions synchronously is discouraged in NodeJS if you are only accessing the filesystem and not a remote target. There is almost no latency and there is no competition for cpu resources in the  node process. With async/await, this has become a no-brainer, for sure, I am just not sure if there is really a need for it. But in the end, I don't mind.

Reply:
My impression is that nodejs is encouraging "async everywhere" *including* filing system; years ago with a very early version of the compiler, I did an interesting experiment using *nix `time` comparing compiler vs generate.py and found that generate.py spent most of it's time waiting for I/O.  The generator does have huge I/O requirements because of it's caching mechanism, but still the CPU bound work was an astonishingly small percentage.  AIUI that I/O would have been exclusively filing system I/O, and compiler in contrast was the exact opposite, i.e. compiler was mostly CPU bound and spent almost no time waiting for I/O.  

That test was to help me decide whether I needed to bother with all that painful callback hell (this is way before promises let alone async/await) and it lead me to conclude that it was worth it.

It is true that the cli commands will not have a lot of I/O and so it's not likely to become a bottleneck, but these things are hard to change retrospectively and async/await make it really easy to implement up front.  IMHO it is up to the calling code to decide whether sync is just fine or async is necessary, and particularly as these are utility methods there is a strong argument for async-ifying them.

Reply:
Ok, you have me convinced :-)